packages/contracts-bedrock/test/L2/L1BlockInterop.t.sol

```diff
@@ -1,52 +1,209 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.15;
 
-import { Test } from "forge-std/Test.sol";
-import { Encoding } from "src/libraries/Encoding.sol";
-import { L1BlockInterop, DependencySetSizeMismatch, NotDepositor, ConfigType } from "src/L2/L1BlockInterop.sol";
+// Testing utilities
+import { CommonTest } from "test/setup/CommonTest.sol";
 
-contract L1BlockInteropTest is Test {
-    L1BlockInterop l1Block;
-    address depositor;
+// Target contract dependencies
+import { L1BlockInterop, ConfigType } from "src/L2/L1BlockInterop.sol";
 
-    function setUp() public {
-        l1Block = new L1BlockInterop();
-        depositor = l1Block.DEPOSITOR_ACCOUNT();
+contract L1BlockInteropTest is CommonTest {
+    /// @notice Thrown when a non-depositor account attempts to set L1 block values.
+    error NotDepositor();
+
+    /// @notice Error when a chain ID is not in the interop dependency set.
+    error NotDependency();
+
+    /// @notice Error when the interop dependency set size is too large.
+    error DependencySetSizeTooLarge();
+
+    /// @notice Error when the chain's chain ID is attempted to be removed from the interop dependency set.
+    error CantRemovedChainId();
+
+    /// @notice Event emitted when the gas paying token is set.
+    event GasPayingTokenSet(address indexed token, uint8 indexed decimals, bytes32 name, bytes32 symbol);
+
+    /// @notice Event emitted when a new dependency is added to the interop dependency set.
+    event DependencyAdded(uint256 indexed chainId);
+
+    /// @notice Event emitted when a dependency is removed from the interop dependency set.
+    event DependencyRemoved(uint256 indexed chainId);
+
+    /// @notice Marked virtual to be overridden in
+    ///         test/kontrol/deployment/DeploymentSummary.t.sol
+    function setUp() public virtual override {
+        super.setUp();
+        vm.etch(address(l1Block), address(new L1BlockInterop()).code);
```

Comment:
We should use the special config flag to deploy the interop contract in the deploy script